verification/src/http_server/handlers/verification/solidity/types.rs

```diff
@@ -18,40 +18,47 @@ pub struct VerificationRequest<T> {
 }
 
 #[derive(Debug, Deserialize, PartialEq)]
-pub struct FlattenedSource {
-    source_code: String,
+pub struct SourcesInput {
+    sources: BTreeMap<PathBuf, String>,
     evm_version: String,
     optimization_runs: Option<usize>,
     contract_libraries: Option<BTreeMap<String, String>>,
 }
 
-impl TryFrom<FlattenedSource> for CompilerInput {
+impl TryFrom<SourcesInput> for CompilerInput {
     type Error = anyhow::Error;
 
-    fn try_from(source: FlattenedSource) -> Result<Self, Self::Error> {
+    fn try_from(input: SourcesInput) -> Result<Self, Self::Error> {
         let mut settings = Settings::default();
-        settings.optimizer.enabled = Some(source.optimization_runs.is_some());
-        settings.optimizer.runs = source.optimization_runs;
-        if let Some(source_libraries) = source.contract_libraries {
-            settings.libraries = Libraries {
-                libs: BTreeMap::from([(PathBuf::from("source.sol"), source_libraries)]),
-            };
+        settings.optimizer.enabled = Some(input.optimization_runs.is_some());
+        settings.optimizer.runs = input.optimization_runs;
+        if let Some(libs) = input.contract_libraries {
+            // we have to know filename for library, we don't know,
+            // so we assume that every file MAY contains all libraries
+            let libs = input
+                .sources
+                .iter()
+                .map(|(filename, _)| (PathBuf::from(filename), libs.clone()))
+                .collect();
+            settings.libraries = Libraries { libs };
         }
-        if source.evm_version != "default" {
+
+        if input.evm_version != "default" {
             settings.evm_version =
-                Some(EvmVersion::from_str(&source.evm_version).map_err(anyhow::Error::msg)?);
+                Some(EvmVersion::from_str(&input.evm_version).map_err(anyhow::Error::msg)?);
         } else {
             // `Settings::default()` sets the value to the latest available evm version (`Some(London)` for now)
             settings.evm_version = None
         }
+
+        let sources: Sources = input
+            .sources
+            .into_iter()
+            .map(|(name, content)| (name, Source { content }))
```

Comment:
It would be great to find method of deserialization to `Source` struct, but i didnt come up with idea